### PR TITLE
Fix check curl exists

### LINKED ISSUE #25: Step reports success even if alert fails.
screen shot 2019-01-03 at 6 49 40 pm

My relevant config:

```
--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -1,4 +1,8 @@
 version: 2.1
+
+orbs:
+  slack: circleci/slack@0.1.15
+
 jobs:
   build:
     docker:
@@ -23,6 +27,8 @@ jobs:
+      - slack/status:
+          fail_only: 'true'
```

### PR DESCRIPTION
**Pull request by piyush-insider**
### Checklist

- [x] All new jobs, commands, executors, parameters have descriptions
- [x] Examples have been added for any significant new features
- [x] README has been updated, if necessary

### Motivation, issues
Fixes #25 

### Description
Using the which utility checking if cURL exists, if it does the code returned is 0 else the code returned is 1 by the command, Which we then check with grep and in case of 1 we exit and echo `curl not installed`

### CODE REVIEW

**Comment by piyush-insider** [comment]
@KyleTryon Thanks for the review. I have added curl as a step in all the three commands. As for the suggestion of having all prerequisite checks in a single step, I think that should be a part of a separate refactor and should not go in along with this change. I'm filing an issue for it separately and will try to pick it up as soon as possible. What do you think?

File: src/commands/approval.yml [hunk 1]
```diff
@@ -41,6 +41,12 @@ steps:
           exit 1
         fi
 
+  - run:
+      name: Provide error if cURL is not installed.
+      command: |
+        which curl > curl_exists; echo $? | grep -q '1' && echo curl not installed && rm curl_exists && exit 1
```
Comment (sachincool) [inline src/commands/approval.yml]: @piyush-insider try implementing this it looks better and without any External file creation. https://scripter.co/check-if-a-command-exists-from-shell-script/

Comment (piyush-insider) [reply src/commands/approval.yml]: Sure. Will take a look at it

**Comment by KyleTryon** [comment]
I have closed issue #25 to reduce the duplicate PR submissions. This PR will be our active branch for this fix.

File: src/commands/approval.yml (hunk 1 above)
Comment (KyleTryon) [reply src/commands/approval.yml]: It looks as if once this change is added we should be ready to merge.

Comment (wyardley) [reply src/commands/approval.yml]: `command -v` should be a good way to do it too, I think.

**Comment by eddiewebb** [comment]
@KyleTryon - I suggest we merge `[skip-ci]` to staging, and we can mass promote all staging additions end of month?

**Pull request merged**